### DISCUSSION TITLE
Delete website Data for a Specified Period

### DISCUSSION THREAD

**Discussion by LahlouMohamed** [umami-software/umami#2789] [Ideas]
**Objective**
Add the ability to delete data for a specified period for Umami users.

**Context**
This feature aims to improve performance management by allowing the deletion of data within specific date ranges. This helps to maintain a lighter and more performant database. Additionally, an import/export feature will be ideal for archiving complete periods before deletion.

**Implementation Details**

1. Database
  Addition of cascading delete functionality to ensure that all data related to a specific period is correctly removed.
  Update of database schemas and necessary migration scripts.
2. API
  Create a new API endpoint that takes two parameters: startDate and endDate.
  Implement the logic for deleting data (event_data) within this date range.
3. Frontend
  Add a user interface to select a date range and initiate the deletion of the corresponding data.
  Display appropriate confirmations and alerts to prevent accidental deletions.
5. Screenshot
<img width="1506" alt="cap" src="https://github.com/umami-software/umami/assets/40507691/7b7ca026-94b1-4cef-9f61-48ebdbfb7033">

<img width="839" alt="Capture d’écran 2024-06-19 à 09 23 24" src="https://github.com/umami-software/umami/assets/40507691/baf56130-7649-4a39-b811-a37e4957337a">

**Comment by aleksandereiken** [comment]
This would be extremely useful to get implemented. I'm a big fan!

**Comment by tschannik** [comment]
Indeed, i'd love a feature like that. It would also be nice if the implemented feature would do the useful database optimization like updating its indices and would run optimize table, right?
